### PR TITLE
Add launchSettings.json by default to our templates

### PR DESCRIPTION
**Pull request by javiercn**
* Adds launchSettings.json to all of our templates.
* Provides an alternative flag to exclude it.
* Adds HTTPS support in launchSettings.json.
* Improves some infrastructure aspects.
    * Prefix all our test projects with AspNet.Template to allow easy
      killing of orphaned processes.
    * Setup an HTTPS certificate automatically for use with tests in the
      same way our tooling does it on customers machines.
    * Switch from OS selected ports to randomly selected ports. This is
      required as we need to know the port to use for HTTPS.
    * Allow disabling browser automation through an environment variable.
    * Setup Firefox and Edge to allow untrusted certificates when using
      Selenium.

### CODE REVIEW

File: test/Templates.Test/Helpers/AspNetProcess.cs [hunk 1]
```diff
@@ -48,9 +61,12 @@ public AspNetProcess(ITestOutputHelper output, string workingDirectory, string p
                     .WaitForExit(assertSuccess: true);
             }
 
+            _httpPort = 5000 + rnd.Next(2000);
+            _httpsPort = 44300 + rnd.Next(1000);
```
Comment (javiercn) [inline test/Templates.Test/Helpers/AspNetProcess.cs]: The plan is for this to go away with https://github.com/aspnet/BasicMiddleware/issues/275

File: test/Templates.Test/Helpers/AspNetProcess.cs [hunk 2]
```diff
@@ -48,9 +61,12 @@ public AspNetProcess(ITestOutputHelper output, string workingDirectory, string p
...
             var envVars = new Dictionary<string, string>
             {
-                { "ASPNETCORE_URLS", "http://127.0.0.1:0" }
+                { "ASPNETCORE_URLS", $"http://localhost:{_httpPort};https://localhost:{_httpsPort}" },
+                { "ASPNETCORE_HTTPS_PORT", $"{_httpsPort}" }
```
Comment (javiercn) [inline test/Templates.Test/Helpers/AspNetProcess.cs]: The plan is for this to go away with aspnet/BasicMiddleware#275

File: src/Microsoft.DotNet.Web.ProjectTemplates/content/WebApi-FSharp/Properties/launchSettings.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        "ASPNETCORE_ENVIRONMENT": "Development",
+        "ASPNETCORE_HTTPS_PORT": "43434"
+      }
+    },
+    "Company.WebApplication1": {
+      "commandName": "Project",
+      "launchBrowser": true,
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development",
+        "ASPNETCORE_HTTPS_PORT": "43434"
```
Comment (javiercn) [inline src/Microsoft.DotNet.Web.ProjectTemplates/content/WebApi-FSharp/Properties/launchSettings.json]: The plan is for this to go away with aspnet/BasicMiddleware#275

File: test/Templates.Test/Helpers/AspNetProcess.cs [hunk 4]
```diff
@@ -48,9 +61,12 @@ public AspNetProcess(ITestOutputHelper output, string workingDirectory, string p
                     .WaitForExit(assertSuccess: true);
             }
 
+            _httpPort = 5000 + rnd.Next(2000);
+            _httpsPort = 44300 + rnd.Next(1000);
             var envVars = new Dictionary<string, string>
             {
-                { "ASPNETCORE_URLS", "http://127.0.0.1:0" }
+                { "ASPNETCORE_URLS", $"http://localhost:{_httpPort};https://localhost:{_httpsPort}" },
```
Comment (javiercn) [inline test/Templates.Test/Helpers/AspNetProcess.cs]: I had to switch the strategy for picking up the ports due to the fact that Kestrel won't allow me to let the OS pick the port when binding to localhost and that we need to bind to localhost for HTTPS requests to work.

Comment (javiercn) [reply test/Templates.Test/Helpers/AspNetProcess.cs]: I'll switch this back to 127.0.0.1:0 once we have aspnet/BasicMiddleware#275 and alternatively we will set the _httpClient.BaseAddress to http://localhost:<<selected-port>>

File: test/Templates.Test/Helpers/WebDriverFactory.cs [hunk 5]
```diff
@@ -24,13 +26,20 @@ public static IWebDriver CreateWebDriver()
             var result = IsAppVeyor ? CreateFirefoxDriver() : CreateEdgeDriver();
             result.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(1);
             return result;
+
+            IWebDriver CreateFirefoxDriver()
+            {
+                return WebDriverFactory.CreateFirefoxDriver(new FirefoxOptions(){
```
Comment (javiercn) [inline test/Templates.Test/Helpers/WebDriverFactory.cs]: I've validated that this works with latest firefox

File: test/Templates.Test/Helpers/WebDriverFactory.cs [hunk 6]
```diff
@@ -24,13 +26,20 @@ public static IWebDriver CreateWebDriver()
...
+                    AcceptInsecureCertificates = true
+                });
+            }
         }
 
         private static IWebDriver CreateEdgeDriver()
             => new EdgeDriver(EdgeDriverService.CreateDefaultService(BinDir));
```
Comment (javiercn) [inline test/Templates.Test/Helpers/WebDriverFactory.cs]: I haven't found a way to ignore SSL certificate errors on Edge, but trusting the certificate on your box makes it work just fine.

File: test/Templates.Test/Helpers/AspNetProcess.cs (hunk 2 above)
Comment (SteveSandersonMS) [reply test/Templates.Test/Helpers/AspNetProcess.cs]: Can you also log a bug to track that this line needs to be removed from the test at that time?

File: src/Microsoft.DotNet.Web.ProjectTemplates/content/WebApi-FSharp/Properties/launchSettings.json (hunk 3 above)
Comment (SteveSandersonMS) [reply src/Microsoft.DotNet.Web.ProjectTemplates/content/WebApi-FSharp/Properties/launchSettings.json]: Likewise, please also reference this line in the bug saying it needs to be removed.

**Comment by SteveSandersonMS** [comment]
Does this add launchSettings.json to all the templates? I couldn't see any indication that it was added to the SPA templates.

**Comment by SteveSandersonMS** [comment]
I know this is marked as [WIP], but just for the sake of tracking, it doesn't look like all the templates are updated consistently. For example, the EmptyWebCSharp template doesn't have any replacers for the 55555 or 43434 values, and the StarterWebCSharp one has a replacer for the 55555 value but not the 43434 value.

File: test/Templates.Test/Helpers/AspNetProcess.cs (hunk 4 above)
Comment (SteveSandersonMS) [reply test/Templates.Test/Helpers/AspNetProcess.cs]: I know there's a relatively low probability of port collisions with this logic, but it wouldn't be much harder to ensure noncollision just by using an incrementing static variable to make it choose ports 5000, 5001, 5002, etc.

File: test/Templates.Test/Helpers/WebDriverFactory.cs (hunk 5 above)
Comment (SteveSandersonMS) [reply test/Templates.Test/Helpers/WebDriverFactory.cs]: Great. Minor nitpick comment - braces look non-C#-ish :)

File: test/Templates.Test/Helpers/WebDriverFactory.cs (hunk 6 above)
Comment (SteveSandersonMS) [reply test/Templates.Test/Helpers/WebDriverFactory.cs]: Would it work to automate clicking the "ignore invalid cert" button in the browser, similar to how it's suggested here? https://github.com/angular/protractor/issues/2377#issuecomment-289920884

**Comment by SteveSandersonMS** [comment]
Overall approach looks great! Please let me know when this is ready for final review.

File: test/Templates.Test/Helpers/WebDriverFactory.cs (hunk 6 above)
Comment (javiercn) [reply test/Templates.Test/Helpers/WebDriverFactory.cs]: @SteveSandersonMS Yeah, I saw that option when I was looking into it, but I wanted to avoid going that route as feels "fragile". The easiest thing to do is to manually trust the cert on your environments if you can.

**Comment by javiercn** [comment]
@SteveSandersonMS Yes, this will add launchSettings.json to the templates

File: test/Templates.Test/Helpers/AspNetProcess.cs (hunk 2 above)
Comment (javiercn) [reply test/Templates.Test/Helpers/AspNetProcess.cs]: Yep will do

File: test/Templates.Test/Helpers/AspNetProcess.cs (hunk 4 above)
Comment (javiercn) [reply test/Templates.Test/Helpers/AspNetProcess.cs]: Yep, i've already seen this approach cause issues, so I'm going to switch to pick a "random base port" and then pick incremental ports with a static variable

**Comment by danroth27** [comment]
Fixes #142

**Comment by javiercn** [comment]
@SteveSandersonMS I updated everything and addressed feedback.

File: test/Templates.Test/Helpers/AspNetProcess.cs (hunk 2 above)
Comment (SteveSandersonMS) [reply test/Templates.Test/Helpers/AspNetProcess.cs]: Just checking this is done (searched, but couldn't find any bug covering removing this from the test yet).

File: build/dependencies.props [hunk 7]
```diff
@@ -4,11 +4,12 @@
   </PropertyGroup>
   <PropertyGroup Label="Package Versions">
     <InternalAspNetCoreSdkPackageVersion>2.1.0-preview1-15576</InternalAspNetCoreSdkPackageVersion>
-    <MicrosoftAspNetCoreAllPackageVersion>2.1.0-preview1-27644</MicrosoftAspNetCoreAllPackageVersion>
+    <MicrosoftAspNetCoreAllPackageVersion>2.1.0-preview1-27644</MicrosoftAspNetCoreAllPackageVersion>
```
Comment (SteveSandersonMS) [inline build/dependencies.props]: Trailing whitespace

File: build/dependencies.props [hunk 8]
```diff
@@ -4,11 +4,12 @@
...
-    <MicrosoftAspNetCoreAllPackageVersion>2.1.0-preview1-27644</MicrosoftAspNetCoreAllPackageVersion>
+    <MicrosoftAspNetCoreAllPackageVersion>2.1.0-preview1-27644</MicrosoftAspNetCoreAllPackageVersion>    
     <MicrosoftAspNetCoreAuthenticationCookiesPackageVersion>2.1.0-preview1-27644</MicrosoftAspNetCoreAuthenticationCookiesPackageVersion>
     <MicrosoftAspNetCoreAuthenticationJwtBearerPackageVersion>2.1.0-preview1-27644</MicrosoftAspNetCoreAuthenticationJwtBearerPackageVersion>
     <MicrosoftAspNetCoreAuthenticationOpenIdConnectPackageVersion>2.1.0-preview1-27644</MicrosoftAspNetCoreAuthenticationOpenIdConnectPackageVersion>
-    <MicrosoftAspNetCoreDiagnosticsEntityFrameworkCorePackageVersion>2.1.0-preview1-27644</MicrosoftAspNetCoreDiagnosticsEntityFrameworkCorePackageVersion>
+    <MicrosoftAspNetCoreCertificatesGenerationSourcesPackageVersion>2.1.0-preview1-27644</MicrosoftAspNetCoreCertificatesGenerationSourcesPackageVersion><MicrosoftAspNetCoreDiagnosticsEntityFrameworkCorePackageVersion>2.1.0-preview1-27644</MicrosoftAspNetCoreDiagnosticsEntityFrameworkCorePackageVersion>
```
Comment (SteveSandersonMS) [inline build/dependencies.props]: Looks like two  lines joined together

File: test/Templates.Test/Helpers/AspNetProcess.cs [hunk 9]
```diff
@@ -8,23 +8,37 @@
...
+using Microsoft.AspNetCore.Certificates.Generation;
+using System.Threading;
 
 namespace Templates.Test.Helpers
 {
     public class AspNetProcess : IDisposable
     {
         private const string DefaultFramework = "netcoreapp2.0";
         private const string ListeningMessagePrefix = "Now listening on: ";
+        private static int Port = 5000 + new Random(Seed: 0).Next(3000);
```
Comment (SteveSandersonMS) [inline test/Templates.Test/Helpers/AspNetProcess.cs]: This doesn't look right. Having a fixed seed means it's always going to return the same sequence. It would be equivalent just to hard-code the specific value returned by `new Random(Seed: 0).Next(3000)` (which is 2178 if I recall correctly).

File: test/Templates.Test/Helpers/WebDriverFactory.cs [hunk 10]
```diff
@@ -24,13 +26,19 @@ public static IWebDriver CreateWebDriver()
             var result = IsAppVeyor ? CreateFirefoxDriver() : CreateEdgeDriver();
             result.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(1);
             return result;
+
+            IWebDriver CreateFirefoxDriver() =>
```
Comment (SteveSandersonMS) [inline test/Templates.Test/Helpers/WebDriverFactory.cs]: Unreachable code (presumbly wasn't meant to be included in the PR)

File: test/Templates.Test/Helpers/WebDriverFactory.cs [hunk 11]
```diff
@@ -24,13 +26,19 @@ public static IWebDriver CreateWebDriver()
...
+                });
         }
 
         private static IWebDriver CreateEdgeDriver()
             => new EdgeDriver(EdgeDriverService.CreateDefaultService(BinDir));
 
-        private static IWebDriver CreateFirefoxDriver()
-            => new FirefoxDriver(FirefoxDriverService.CreateDefaultService(BinDir));
+        private static IWebDriver CreateFirefoxDriver(FirefoxOptions options)
+            => new FirefoxDriver(FirefoxDriverService.CreateDefaultService(BinDir), options, TimeSpan.FromSeconds(1));
```
Comment (SteveSandersonMS) [inline test/Templates.Test/Helpers/WebDriverFactory.cs]: Can this change be reverted? It doesn't like it's used except by the unreachable code mentioned above.

File: test/Templates.Test/MvcTemplateTest.cs [hunk 12]
```diff
@@ -13,18 +13,14 @@ public MvcTemplateTest(ITestOutputHelper output) : base(output)
         {
         }
 
-        [ConditionalTheory]
+        [ConditionalFact]
         [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
-        [InlineData(null)]
-        [InlineData("F#")]
```
Comment (SteveSandersonMS) [inline test/Templates.Test/MvcTemplateTest.cs]: It looks like an unrelated change has ended up in this PR. Can we revert the F# change?

File: test/Templates.Test/Templates.Test.csproj [hunk 13]
```diff
@@ -2,10 +2,12 @@
 
   <PropertyGroup>
     <TargetFramework>netcoreapp2.0</TargetFramework>
+    <DefineConstants>$(DefineConstants);XPLAT</DefineConstants>
```
Comment (SteveSandersonMS) [inline test/Templates.Test/Templates.Test.csproj]: What is this used for? Was it meant to be included in the PR?

**Comment by SteveSandersonMS** [comment]
Also the AppVeyor tests aren't looking healthy: https://ci.appveyor.com/project/aspnetci/templating/build/1.0.53/tests

I know there's been quite a bit of red on AppVeyor lately but at least the majority of that is: (a) legitimate code issues that need to be fixed ([example](https://ci.appveyor.com/project/aspnetci/templating/build/1.0.52/tests)), and (b) different from the errors showing up for this PR.

File: test/Templates.Test/Templates.Test.csproj (hunk 13 above)
Comment (javiercn) [reply test/Templates.Test/Templates.Test.csproj]: Yep. This is to avoid a dependency on System.Security.Cryptography.Cng

Comment (SteveSandersonMS) [reply test/Templates.Test/Templates.Test.csproj]: I guess I'm missing where it's used then. Is it a magic value that something in the compilation chain is looking for (and if so, what)?

Comment (javiercn) [reply test/Templates.Test/Templates.Test.csproj]: It's a define for an #if XPLAT that we use on the shared sources library that we bring in to setup an HTTPS certificate for the tests

File: test/Templates.Test/MvcTemplateTest.cs (hunk 12 above)
Comment (javiercn) [reply test/Templates.Test/MvcTemplateTest.cs]: It impacts the ability for the tests to pass. See the email thread we have about this for details, I can do it in a separate PR if you want me to

File: test/Templates.Test/Helpers/WebDriverFactory.cs (hunk 10 above)
Comment (javiercn) [reply test/Templates.Test/Helpers/WebDriverFactory.cs]: This gets called here https://github.com/aspnet/templating/pull/167/files/aef1328ed6ff41aeaace505f954798ce687acdab#diff-7765183c367cb5141c05166cc20ed8d7L24

Which in turn calls

https://github.com/aspnet/templating/pull/167/files/aef1328ed6ff41aeaace505f954798ce687acdab#diff-7765183c367cb5141c05166cc20ed8d7R40

File: test/Templates.Test/Templates.Test.csproj (hunk 13 above)
Comment (SteveSandersonMS) [reply test/Templates.Test/Templates.Test.csproj]: OK thanks!

File: test/Templates.Test/Helpers/AspNetProcess.cs (hunk 2 above)
Comment (javiercn) [reply test/Templates.Test/Helpers/AspNetProcess.cs]: I was planning to file the bugs at merging time. I'll file them right away

File: test/Templates.Test/Helpers/AspNetProcess.cs (hunk 9 above)
Comment (javiercn) [reply test/Templates.Test/Helpers/AspNetProcess.cs]: Fair enough, I'll remove the seed part (should pick a different seed each time I hope).
I just want the base port to be random to minimize the change of port conflicts

File: test/Templates.Test/Helpers/AspNetProcess.cs (hunk 2 above)
Comment (javiercn) [reply test/Templates.Test/Helpers/AspNetProcess.cs]: > It looks like the HTTPS middleware bits are reintroduced with this PR. I just want to check that this is intentional and this new HTTPS functionality is definitely all ready to ship. Otherwise can we have that as a separate PR?

Yep, I spoke to @DamianEdwards about it yesterday and we've gone through the pertinent reviews and we are all happy so far with it. @danroth27 can confirm

**Comment by javiercn** [comment]
> It looks like the HTTPS middleware bits are reintroduced with this PR. I just want to check that this is intentional and this new HTTPS functionality is definitely all ready to ship. Otherwise can we have that as a separate PR?

Yep, I spoke to @DamianEdwards about it yesterday and we've gone through the pertinent reviews and we are all happy so far with it. @danroth27 can confirm

File: test/Templates.Test/Helpers/WebDriverFactory.cs (hunk 10 above)
Comment (SteveSandersonMS) [reply test/Templates.Test/Helpers/WebDriverFactory.cs]: Oh I see. I misread it.

But it is really strange to declare this as an inline function like this, when there's a same-named function in the outer scope, which in turn no longer gets used except by this inline function.

To make it a bit more obvious, could you just change the existing `CreateFirefoxDriver` method so that it always sets `AcceptInsecureCertificates = true`, without needing the new local function?

**Comment by SteveSandersonMS** [comment]
> Yep, I spoke to @DamianEdwards about it yesterday and we've gone through the pertinent reviews and we are all happy so far with it

Great - thanks for confirming!

**Comment by javiercn** [comment]
> Also the AppVeyor tests aren't looking healthy: https://ci.appveyor.com/project/aspnetci/templating/build/1.0.53/tests

I'm not sure whats going on here. The server seems to start just fine. I tried to log into AppVeyor but somehow I don't have permissions for re-triggering the build?

I'm going to try and investigate locally. Can it be related to any of the servers that the spa middleware spin up not playing nice with HTTPS? 

I'm just saying this because some of the tests related to selenium seem to pass (MVC Individual user accounts for example)

 Time Elapsed 00:00:11.42
 Running ASP.NET application...
 ==> C:\Users\appveyor\.dotnet\x64\dotnet.exe exec bin/Debug/netcoreapp2.0/AspNet.Template.cecb0eaffa8244ad91ce10c360145a8e.dll [C:\projects\templating\test\Templates.Test\bin\Release\netcoreapp2.0\TestTemplates\AspNet.Template.cecb0eaffa8244ad91ce10c360145a8e]
 Waiting until ASP.NET application is accepting connections...
 info: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[0]
       User profile is available. Using 'C:\Users\appveyor\AppData\Local\ASP.NET\DataProtection-Keys' as key repository and Windows DPAPI to encrypt keys at rest.
 Hosting environment: Development
 Content root path: C:\projects\templating\test\Templates.Test\bin\Release\netcoreapp2.0\TestTemplates\AspNet.Template.cecb0eaffa8244ad91ce10c360145a8e
 Now listening on: http://localhost:7195
 Now listening on: https://localhost:7196
 Application started. Press Ctrl+C to shut down.
 Detected that ASP.NET application is accepting connections on: http://localhost:7195
 info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
       Request starting HTTP/1.1 GET http://localhost:7195/  
 info: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[1]
       Executing action method AspNet.Template.cecb0eaffa8244ad91ce10c360145a8e.Controllers.HomeController.Index (AspNet.Template.cecb0eaffa8244ad91ce10c360145a8e) with arguments ((null)) - ModelState is Valid
 info: Microsoft.AspNetCore.NodeServices[0]
       webpack built e2dbeb095feb53db6b58 in 4114ms
 info: Microsoft.AspNetCore.Mvc.ViewFeatures.ViewResultExecutor[1]
       Executing ViewResult, running view at path /Views/Home/Index.cshtml.
 info: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[2]
       Executed action AspNet.Template.cecb0eaffa8244ad91ce10c360145a8e.Controllers.HomeController.Index (AspNet.Template.cecb0eaffa8244ad91ce10c360145a8e) in 4131.3481ms
 Opening browser at http://localhost:7195/...
 info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
       Request finished in 4379.0587ms 200 text/html; charset=utf-8
1512520779992	addons.xpi	WARN	Error parsing extensions state: [Exception... "Component returned failure code: 0x80520012 (NS_ERROR_FILE_NOT_FOUND) [amIAddonManagerStartup.readStartupData]"  nsresult: "0x80520012 (NS_ERROR_FILE_NOT_FOUND)"  location: "JS frame :: resource://gre/modules/addons/XPIProvider.jsm :: loadExtensionState :: line 1554"  data: no] Stack trace: loadExtensionState()@resource://gre/modules/addons/XPIProvider.jsm:1554 < getInstallState()@resource://gre/modules/addons/XPIProvider.jsm:1589 < checkForChanges()@resource://gre/modules/addons/XPIProvider.jsm:3109 < startup()@resource://gre/modules/addons/XPIProvider.jsm:2188 < callProvider()@resource://gre/modules/AddonManager.jsm:269 < _startProvider()@resource://gre/modules/AddonManager.jsm:739 < startup()@resource://gre/modules/AddonManager.jsm:906 < startup()@resource://gre/modules/AddonManager.jsm:3090 < observe()@jar:file:///C:/Program%20Files%20(x86)/Mozilla%20Firefox/omni.ja!/components/addonManager.js:65
1512520781317	Marionette	INFO	Enabled via --marionette
1512520784766	Marionette	INFO	Listening on port 2140
1512520785013	Marionette	WARN	TLS certificate errors will be ignored for this session
1512520785082	Marionette	DEBUG	Register listener.js for window 4294967297
[xUnit.net 00:03:01.3856741]     Templates.Test.SpaTemplateTest.KnockoutTemplateTest.KnockoutTemplate_Works_NetCore [FAIL]
[xUnit.net 00:03:01.3858096]       OpenQA.Selenium.WebDriverException : The HTTP request to the remote WebDriver server for URL http://localhost:2227/session timed out after 1 seconds.
[xUnit.net 00:03:01.3859147]       ---- System.Net.WebException : The operation has timed out.
[xUnit.net 00:03:01.3859799]       Stack Trace:
[xUnit.net 00:03:01.3860445]            at OpenQA.Selenium.Remote.HttpCommandExecutor.MakeHttpRequest(HttpRequestInfo requestInfo)
[xUnit.net 00:03:01.3860921]            at OpenQA.Selenium.Remote.HttpCommandExecutor.Execute(Command commandToExecute)
[xUnit.net 00:03:01.3861695]            at OpenQA.Selenium.Remote.DriverServiceCommandExecutor.Execute(Command commandToExecute)
[xUnit.net 00:03:01.3862175]            at OpenQA.Selenium.Remote.RemoteWebDriver.Execute(String driverCommandToExecute, Dictionary`2 parameters)
[xUnit.net 00:03:01.3862648]            at OpenQA.Selenium.Remote.RemoteWebDriver.StartSession(ICapabilities desiredCapabilities)
[xUnit.net 00:03:01.3863132]            at OpenQA.Selenium.Remote.RemoteWebDriver..ctor(ICommandExecutor commandExecutor, ICapabilities desiredCapabilities)
[xUnit.net 00:03:01.3863918]         C:\projects\templating\test\Templates.Test\Helpers\WebDriverFactory.cs(41,0): at Templates.Test.Helpers.WebDriverFactory.CreateFirefoxDriver(FirefoxOptions options)
[xUnit.net 00:03:01.3865072]         C:\projects\templating\test\Templates.Test\Helpers\WebDriverFactory.cs(26,0): at Templates.Test.Helpers.WebDriverFactory.CreateWebDriver()
[xUnit.net 00:03:01.3866128]         C:\projects\templating\test\Templates.Test\Helpers\AspNetProcess.cs(129,0): at Templates.Test.Helpers.AspNetProcess.VisitInBrowser()
[xUnit.net 00:03:01.3867164]         C:\projects\templating\test\Templates.Test\SpaTemplateTest\SpaTemplateTestBase.cs(50,0): at Templates.Test.SpaTemplateTest.SpaTemplateTestBase.TestApplication(String targetFrameworkOverride, Boolean publish)
[xUnit.net 00:03:01.3868367]         C:\projects\templating\test\Templates.Test\SpaTemplateTest\SpaTemplateTestBase.cs(43,0): at Templates.Test.SpaTemplateTest.SpaTemplateTestBase.SpaTemplateImpl(String targetFrameworkOverride, String template)
[xUnit.net 00:03:01.3869458]         ----- Inner Stack Trace -----
[xUnit.net 00:03:01.3869939]            at System.Net.HttpWebRequest.GetResponse()
[xUnit.net 00:03:01.3870396]            at OpenQA.Selenium.Remote.HttpCommandExecutor.MakeHttpRequest(HttpRequestInfo requestInfo)

File: test/Templates.Test/Helpers/AspNetProcess.cs (hunk 2 above)
Comment (javiercn) [reply test/Templates.Test/Helpers/AspNetProcess.cs]: Filed https://github.com/aspnet/templating/issues/181

**Comment by javiercn** [comment]
@SteveSandersonMS All the feedback has been addressed and the selenium tests are Happy. I'm optimistically merging the PR and I will address any further feedback you might have.
